pool/app/session/entity.ml

```diff
@@ -246,6 +246,30 @@ module Public = struct
     |> CCList.map (fun (_, (p, fs)) -> p, CCList.sort compare_start fs)
     |> CCList.sort (fun (f1, _) (f2, _) -> compare_start f1 f2)
   ;;
+
+  let group_and_sort_keep_followups sessions =
```

Comment:
What is the difference to "group_and_sort"?

How about `group_and_sort ||> CCList.flatten`?